examples/planning/forecast_reach.py

```diff
@@ -0,0 +1,277 @@
+    """
+    reach_plan_service = client.get_service('ReachPlanService', version='v2')
+    response = reach_plan_service.list_plannable_products(
+            plannable_location_id=_string_value(client, location_id))
+    print('Plannable Products for Location ID {}'.format(location_id))
+    print(response)
+
+
+def _request_reach_curve(
+        client, customer_id, product_mix, location_id, currency_code):
+    """Creates a sample request for a given product mix.
+
+    Args:
+      client: A google.ads.google_ads.client.GoogleAdsClient instance.
+      customer_id: The customer ID for the reach forecast.
+      product_mix: The product mix for the reach forecast.
+      location_id: Location ID from somewhere returned by ListPlannableLocations.
+      currency_code: Three-character ISO 4217 currency code.
+    """
+    reach_request = client.get_type('GenerateReachForecastRequest', version='v2')
+
+    reach_request.customer_id = customer_id
+
+    duration = reach_request.campaign_duration
+    duration.duration_in_days.value = 30
+
+    targeting = reach_request.targeting
+    targeting.plannable_location_id.value = location_id
+    targeting.age_range = client.get_type(
+            'ReachPlanAgeRangeEnum', version='v2').AGE_RANGE_18_65_UP
+
+    genders = targeting.genders
+    gender_types = [
+            client.get_type('GenderTypeEnum', version='v2').FEMALE,
+            client.get_type('GenderTypeEnum', version='v2').MALE,
+    ]
+    for gender_type in gender_types:
+        gender = client.get_type('GenderInfo', version='v2')
+        gender.type = gender_type
+        genders.append(gender)
+
+    devices = targeting.devices
+    device_types = [
+            client.get_type('DeviceEnum', version='v2').DESKTOP,
+            client.get_type('DeviceEnum', version='v2').MOBILE,
+            client.get_type('DeviceEnum', version='v2').TABLET,
+    ]
+    for device_type in device_types:
+        device = client.get_type('DeviceInfo', version='v2')
+        device.type = device_type
+        devices.append(device)
+
+    reach_plan_service = client.get_service('ReachPlanService', version='v2')
+    response = reach_plan_service.generate_reach_forecast(
+            customer_id=customer_id,
+            currency_code=_string_value(client, currency_code),
+            campaign_duration=duration,
+            cookie_frequency_cap=_int_32_value(client, 0),
+            min_effective_frequency=_int_32_value(client, 3),
+            targeting=targeting,
+            planned_products=product_mix
+    )
+
+    print('Currency, Cost, On-Target Reach, On-Target Imprs, Total Reach,'
+          ' Total Imprs, Products')
+    for point in response.reach_curve.reach_forecasts:
+        product_splits = []
+        for p in point.forecasted_product_allocations:
+            product_splits.append({
+                    p.plannable_product_code.value:
+                        p.budget_micros.value / 1.0e6})
```

Comment:
better practice to make 1.0e6 a const, ie. at the top define
MICROS_TO_CURRENCY = 1.0e6

as different local currencies may have different conversions.

Reply:
Since we use this to convert back / forth, just named it `ONE_MILLION`.